starknet_py/net/account/account.py

```diff
@@ -79,26 +83,31 @@ def __init__(
         :param key_pair: Key pair that will be used to create a default `Signer`.
         :param chain: ChainId of the chain used to create the default signer.
         """
-        if chain is None and signer is None:
-            raise ValueError("One of chain or signer must be provided.")
-
         self._address = parse_address(address)
         self._client = client
 
-        if signer is not None and key_pair is not None:
-            raise ValueError("Arguments signer and key_pair are mutually exclusive.")
-
-        if signer is None:
-            if key_pair is None:
+        if signer is not None:
+            if chain is not None:
+                warnings.warn("Arguments signer and chain are mutually exclusive.")
+            if key_pair is not None:
                 raise ValueError(
-                    "Either a signer or a key_pair must be provided in Account constructor."
+                    "Arguments signer and key_pair are mutually exclusive."
                 )
 
-            chain = chain_from_network(net=client.net, chain=chain)
-            signer = StarkCurveSigner(
-                account_address=self.address, key_pair=key_pair, chain_id=chain
+            self.signer = signer
+            return
+
+        if key_pair is None:
+            raise ValueError(
+                "Either a signer or a key_pair must be provided in Account constructor."
             )
-        self.signer: BaseSigner = signer
+
+        if chain is None:
+            chain = StarknetChainId.from_network(net=client.net)
+
+        self.signer = StarkCurveSigner(
+            account_address=self.address, key_pair=key_pair, chain_id=chain
+        )
```

Comment:
This logic has gotten quite complex now. I think it's a good time to separate parts of this into helper functions so its easier to follow.